lib/sequelize.js

```diff
@@ -893,9 +893,12 @@ module.exports = (function() {
         }
       });
 
+      // Iterate daos and sync.  Collect results into array
       return Promise.reduce(daos, function(total, dao) {
-        return dao.sync(options);
-      }, null);
+        return total.push(dao.sync(options));
```

Comment:
Hey Mick, thanks.  Fixed this a different way, but I like your syntax better.  Just a sec and I'll commit.